ResourceStatusSystem/Agent/EmailAgent.py

```diff
@@ -0,0 +1,149 @@
+''' EmailAgent
+
+  This agent sends an email every 30 minutes with aggregated actions taken for each site
+
+'''
+
+
+import os
+import json
+from datetime import datetime
+from DIRAC                                                       import S_OK, S_ERROR
+from DIRAC.Core.Base.AgentModule                                 import AgentModule
+from DIRAC.ResourceStatusSystem.Utilities                        import RssConfiguration
+from DIRAC.Interfaces.API.DiracAdmin                             import DiracAdmin
+
+__RCSID__ = '$Id: $'
+
+AGENT_NAME = 'ResourceStatus/EmailAgent'
+
+class EmailAgent( AgentModule ):
+
+  def __init__( self, *args, **kwargs ):
+
+    AgentModule.__init__( self, *args, **kwargs )
+    self.diracAdmin = DiracAdmin()
+
+    self.default_value = None
+    self.dirac_path = None
+    self.dirac_path = None
+    self.cacheFile = None
+
+  def initialize( self ):
+    ''' EmailAgent initialization
+    '''
+
+    self.default_value = '/home/smiras/DIRAC_Development/test_dir/'
+    #self.default_value = '/opt/dirac/pro/work/ResourceStatus/'
+    self.dirac_path = os.getenv('DIRAC', self.default_value)
+    self.cacheFile = self.dirac_path + 'cache.json'
+
+    return S_OK()
+
+  def execute( self ):
+
+    #if the file exists and it is not empty
+    if (os.path.isfile(self.cacheFile)) and (os.stat(self.cacheFile).st_size > 0):
+
+      time = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+
+      #load the file
+      with open(self.cacheFile) as f:
+        new_dict = json.load(f)
+
+      #read all the name elements of a site
+      for site in new_dict:
+        subject = "RSS actions taken for " + site
+        body = self._emailBodyGenerator(self.cacheFile, site)
```

Comment:
If I read it only once (during the initialization of the agent for example) and pass the dictionary around later that dictionary is going to be kept only in the computer's temporary memory, right?

I was thinking that if the `EmailAgent` shuts down unexpectedly (which can be a rare case I admit) the precious changes of the dictionary will not be lost, that's why I'm writing to it again and again.

Correct me if I have misunderstood anything.